histomicstk/utils/sample_pixels.py

```diff
@@ -70,7 +70,15 @@ def sample_pixels(slide_path, sample_fraction=None, magnification=None,
         format=large_image.tilesource.TILE_FORMAT_NUMPY,
         scale=scale_lres
     )
-    im_lres = im_lres[:, :, :3]
+
+    # TODO - add code for single channel image
+    print('n\\ >>Low res image shape', im_lres.shape)
+    if len(im_lres.shape) <= 2 or im_lres.shape[2] == 1:
+        im_lres = np.dstack((im_lres, im_lres, im_lres))
+        if flag_color_inversion:
```

Comment:
I agree, the best way to go will be to apply the inversion regardless of the number of bands. I have updated the code to have the inversion for all kinds of images. I haven't removed any TODO yet... since there is one more route the images take when is_wsi is not triggered. I will need to update that code also to have a complete effect for the image inversion.